x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/job_creator.ts

```diff
@@ -230,6 +235,48 @@ export class JobCreator {
     }
   }
 
+  public get sparseData(): boolean {
+    return this._sparseData;
+  }
+
+  public set sparseData(sparseData: boolean) {
+    this._sparseData = sparseData;
+    this._updateSparseDataDetectors();
+  }
+
+  private _updateSparseDataDetectors() {
+    this._detectors.forEach((d, i) => {
+      if (this._aggs[i].id === ML_JOB_AGGREGATION.COUNT) {
+        d.function = this._sparseData
+          ? ML_JOB_AGGREGATION.NON_ZERO_COUNT
+          : ML_JOB_AGGREGATION.COUNT;
+      }
+      if (this._aggs[i].id === ML_JOB_AGGREGATION.HIGH_COUNT) {
```

Comment:
Could this and the other `if`s be `else if`s so it doesn't go through every `if`?

Reply:
this whole thing can be written so much better.
the whole check can be a switch inside a single `if (this._sparseData === true) {` check.